### PR TITLE
Improved Logging and Fixed Fallback bug

### LINKED ISSUE #8: Improving logging and fixed fallback bug
**Fallback notification bug**
>Upon displaying of notification it cannot display the email attached and hence goes for fallback  which says something not so helpful.
Hence I improved the fallback text.
Now it says  ----> _An email by <email_sender>_

 **Improved logging for the server**
>This is to simplify debugging

**I am submitting a pull request for the above mentioned issues** 

**Added log for status code of WEBHOOK_URL**

### PR DESCRIPTION
**Pull request by berserker1**
Closes #8 .
@OrkoHunter I have tested this with a workspace I created.

Parameters
_parameters_
Sending the following data to _webhook_url_
_data_
Exit with status code  _status_code_

**Enough whitespace is there to distinguish between each transition**

### CODE REVIEW

**Comment by berserker1** [comment]
@OrkoHunter You can test on your own or I can also invite to the workspace for that

File: .gitignore [hunk 1]
```diff
@@ -113,3 +113,6 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+
+#for vscode
+.vscode
```
Comment (OrkoHunter) [inline .gitignore:118]: Configure your editor
1. To add a new line at the end of the file while saving.
2. Trim any extra lines at the end of files.
3. Trim any trailing whitespace.

Comment (OrkoHunter) [inline .gitignore:118]: Please add some extension to auto-fix pep8 issues or atleast highlight them in your editor itself.

Comment (OrkoHunter) [inline .gitignore:118]: Please rename this Pull Request to what it does.

Comment (OrkoHunter) [inline .gitignore:118]: Minor: If you add `Closes #8` in the PR description, merging this PR will automatically close that issue.

**Review by OrkoHunter — CHANGES_REQUESTED** [review]
Great work @berserker1!

There are some minor comments I left in my review. Can you please squash the commits into 1 or 2 meaningful commits?

Also, if you say that you have tested it in your workspace, then I believe you.

File: .gitignore (hunk 1 above)
Comment (berserker1) [reply .gitignore:118]: what would you suggest for vscode?

Comment (berserker1) [reply .gitignore:118]: Can we rename this or should I make a new PR?

**Comment by berserker1** [comment]
> Great work @berserker1!
> 
> There are some minor comments I left in my review. Can you please squash the commits into 1 or 2 meaningful commits?
> 
> Also, if you say that you have tested it in your workspace, then I believe you.

Can I squash this here only or should I make a new PR?

File: .gitignore (hunk 1 above)
Comment (OrkoHunter) [reply .gitignore:118]: Aah sorry! My comment was for the whole PR and not for this line.

Comment (OrkoHunter) [reply .gitignore:118]: You can rename this with the edit button above.

**Comment by OrkoHunter** [comment]
> Can I squash this here only or should I make a new PR?

Do not make a new PR. Experiment here!

The way I do is that I soft reset my commits of the branch, individually select the files which make sense and then commit them.

```
git reset --soft HEAD~n  # where n is the number of commits you made
git add <bunch of files>
git commit -S -m "Meaningful Commit message 1"
git add <bunch of files>
git commit -S -m "Meaningful Commit message 1"
git push -f origin <your branch>
```

**Comment by berserker1** [comment]
Cool

**Review by xypnox — APPROVED** [review]
LGTM :+1:

File: app.py [hunk 2]
```diff
@@ -22,22 +22,29 @@ def validate(params):
...
     else:
+        if not app_id:
+            print("APP_ID is not right!")
+        if not token:
+            print("TOKEN is not right!")
+        if not team:
+            print("TEAM_ID is not right!")
+        if not channel:
+            print("USLACKBOT channel is not right!")
+        print("\n\n\n")
```
Comment (xypnox) [inline app.py:33]: More pythonic ways for this section can be implemented
https://stackoverflow.com/questions/6581190/check-if-one-of-variables-is-set-to-none

Although we won't get which variable was wrong

File: .gitignore (hunk 1 above)
Comment (berserker1) [reply .gitignore:118]: No Like I mean any extension for pep8 in vscode

File: app.py (hunk 2 above)
Comment (berserker1) [reply app.py:33]: No so like When I entered TEAM_ID wrong the validate params function was not working
I found it saying this variable to be false

**Comment by xypnox** [comment]
In that case it is perfectly fine!

File: .gitignore (hunk 1 above)
Comment (berserker1) [reply .gitignore:118]: Should I write Closes <#issue_no> only in the description or writing along some other text will also work?

Comment (berserker1) [reply .gitignore:118]: Done all three.

Comment (OrkoHunter) [reply .gitignore:118]: I think GitHub looks for `Closes #no` in the description. You can write other text in it too.

Comment (OrkoHunter) [reply .gitignore:118]: Yeah, you can look for `pycodestyle` in the VSCode marketplace. There are others as well, which you can search for.

**Comment by berserker1** [comment]
@OrkoHunter is this fit for merge now?

**Comment by OrkoHunter** [comment]
Yep. Go ahead!

**Pull request merged**